habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -137,11 +137,18 @@ def reset(self, episode: Episode, fetch_observations: bool = True):
         self._done = False
         self._cur_episode_step = 0
         if fetch_observations:
-            return self._get_observations(episode)
+            return self._get_first_observations(episode)
         else:
             return None
 
-    def _get_observations(self, episode):
+    def _get_first_observations(self, episode):
+        """
+        Retrieve the first observations after a reset.
+        Should only be called after/within a reset call.
+        Tries to updates the robot's sensor position before
+        retrieving observations.
+        """
+        self._sim.maybe_update_robot()
```

Comment:
Why robot update haven't happen during sim.reset() call in line L:125 and in L:130 `self._set_robot_start(agent_idx)`?

Reply:
I tried with the navigation tasks and this issue did not appear. I think the reason why reset is not called in the same way in navigation and in rearrange is because of the task planning. You no not always want to reset after a pick task because there is a subsequent place task in the planner. I think @ASzot will have more context about this.

Reply:
The robot position is actually set in the task on line 130 of habitat/tasks/rearrange/rearrange_task.py. So it is independent of the sim reset.

Reply:
We have same method for Navigation tasks: 

https://github.com/facebookresearch/habitat-lab/blob/00ee28f8339a0b20966f12456728676256ebc246/habitat/sims/habitat_simulator/habitat_simulator.py#L375-L387

Yes, that doesn't cove usecase when episode continues as new task.